bpfman-operator/apis/v1alpha1/cluster_tc_program_types.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// All fields are required unless explicitly marked optional
+// +kubebuilder:validation:Required
+package v1alpha1
+
+// +kubebuilder:validation:Enum=unspec;ok;reclassify;shot;pipe;stolen;queued;repeat;redirect;trap;dispatcher_return
+type TcProceedOnValue string
+
+// ClTcProgramInfo defines the tc program details
+type ClTcProgramInfo struct {
+	// The list of points to which the program should be attached.  The list items
+	// are optional and may be udated after the bpf program has been loaded
+	// +optional
+	// +kubebuilder:default:={}
+	Links []ClTcAttachInfo `json:"links"`
+}
+
+type ClTcAttachInfo struct {
+	// interfaceSelector to determine the network interface (or interfaces)
+	InterfaceSelector InterfaceSelector `json:"interfaceSelector"`
+
+	// containers identifies the set of containers in which to attach the eBPF
+	// program. If Containers is not specified, the BPF program will be attached
+	// in the root network namespace.
+	// +optional
+	Containers *ClContainerSelector `json:"containers"`
+
+	// direction specifies the direction of traffic the tc program should
+	// attach to for a given network device.
+	// +kubebuilder:validation:Enum=ingress;egress
+	Direction string `json:"direction"`
```

Comment:
For an enum, this should use a type alias and have constants.

Reply:
This enum also passed through so it has to stay as "ingress" and "egress" can't change to "Ingress", "Egress", I will ad enum type

Reply:
Your API doesn't have to be identical to an upstream API, just convertible. 

There is nothing to stop you mapping Egress and Ingress to their upstream counterparts when you convert the configuration over.

Reply:
I fixed it and did the mapping internally